### PR TITLE
subversion: remove java bindings

### PR DESCRIPTION
**Pull request by fxcoudert**
A problem that is encountered in bottling is that we have some formulas that require `subversion` to fetch the source, and `subversion` has been made dependent on `openjdk`, which is a beast and often broken for new macOS releases.

Originally, subversion only had an option for the java bindings, and somewhere along the way it was made a build dependency. But it is annoying, and not core to its functionality. I’d like to propose that we remove this dependency. Especially since the formula is used by Homebrew itself.

If there is user demand, we could add the bindings in a separate formula.

### CODE REVIEW

**Comment by Bo98** [comment]
I agree. Though people do use it: https://github.com/subclipse/subclipse/wiki/JavaHL-on-OSX

> If there is user demand, we could add the bindings in a separate formula.

We should do this if it is possible to do this. MacPorts has a `subversion-javahlbindings`.

**Comment by fxcoudert** [comment]
Proof of concept, I do not use java so don't know how to write a test. Help is welcome.

File: Formula/subversion-javahl.rb [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+    ENV.deparallelize
+    system "make", "javahl"
+    system "make", "install-javahl"
+  end
+
+  def caveats
+    <<~EOS
+      You may need to link the Java bindings into the Java Extensions folder:
+        sudo mkdir -p /Library/Java/Extensions
+        sudo ln -s #{HOMEBREW_PREFIX}/lib/libsvnjavahl-1.dylib /Library/Java/Extensions/libsvnjavahl-1.dylib
```
Comment (SMillerDev) [inline Formula/subversion-javahl.rb:176]:
```suggestion
        sudo ln -s #{HOMEBREW_PREFIX}/lib/libsvnjavahl-#{version.major}.dylib /Library/Java/Extensions/libsvnjavahl-#{version.major}.dylib
```

**Review by SMillerDev — DISMISSED** [review]
Makes sense to me

File: Formula/subversion-javahl.rb (hunk 1 above)
Comment (fxcoudert) [reply Formula/subversion-javahl.rb:176]: I think that's being very optimistic about subversion development. Also, I have no idea if they ever make a version 2, whether the javahl libraries would be affected as well?

**Comment by fxcoudert** [comment]
That sucks. The bindings library does not know where to find the subversion libraries, because it was built in its own path.

**Comment by Bo98** [comment]
I had a look at this.

This works by building subversion linking to that, and then only installing the JavaHL bits. That sort of works if the formulae are aligned. It's not great, and a bit fragile, but it can work.

Unsurprisingly this assumes everything is installed into the same prefix, which is not the case for Homebrew. So we might need to just hack replace the paths with ruby-macho.

If there's a way to link via rpaths, that would also work. I'm not sure if there is or not.

The alternative would be replacing `build.conf` with on that defines these dependencies as external. It's a patch job, but does mean proper linking. Maybe too big of a patch job to pursue though.

**Comment by fxcoudert** [comment]
I've looked at configuring with the main subversion path, and then installing through a temporary directory (using `DESTDIR`) then copying the files over. It does not work, because the makefile is badly written and some parts do not obey `DESTDIR`.

At this stage, I think this the formula is too much of a hack already, so I would recommend losing the java part entirely.

**Comment by SMillerDev** [comment]
I'm fine dropping the bindings for maintainer convenience too, but can you make a test that fails if they get added back (or a Java dependency gets added) so we don't mistakenly allow them again.

File: Formula/subversion-javahl.rb (hunk 1 above)
Comment (SMillerDev) [reply Formula/subversion-javahl.rb]: 😄 I just think it looked nicer

**Comment by Bo98** [comment]
I would've been fine with a hack, but I'm very reluctant rushing a breaking change that we know will cause disruption, all just because of the heat of Ventura bottling.

The immediate issue with OpenJDK is caused by this being after the tcp.h include when it needs to be before: https://github.com/openjdk/jdk/blob/9a40b76ac594f5bd80e74ee906af615f74f9a41a/src/jdk.net/macosx/native/libextnet/MacOSXSocketOptions.c#L34. If you want to rush something through, it's trying that.

I recognise we need to do better for the future however as OpenJDK as a dependency to something `brew` itself pulls in is not good.

**Comment by MikeMcQuaid** [comment]
> I would've been fine with a hack, but I'm very reluctant rushing a breaking change that we know will cause disruption, all just because of the heat of Ventura bottling. We've caused far too much disruption from other changes recently.

Agreed.

**Comment by fxcoudert** [comment]
For me, it's not the bottling rush, it's really that we're taking this opportunity to reconsider a previous decision. We've be moving away from heavy dependency trees that are burdened by things few users actually use. I think this is a good example of that. And made even stricter by the fact that we use it internally.

(The experience is even worse for our users who build from source. They have to build openjdk to get a simple svn source? Weird…)

**Comment by fxcoudert** [comment]
> I recognise we need to do better for the future however as OpenJDK as a dependency to something brew itself pulls in is not good.

This is my point. I am not planning on merging this without debate, but I want also not to forget to have this discussion and find a solution. For now I see:

1. a keg-only, binding-free minimalist svn (`subversion-internal`) for brew use (but we try to avoid doing this way, usually)
2. make a java-enabled formula that also ships the subversion library, but is keg-only so it doesn't conflict
3. someone figures out how the way to hack it (but it will be ugly)
4. removing the bindings and see what the user demand is

Again, this was an option before, and it was not used a lot.

**Comment by MikeMcQuaid** [comment]
> * someone figures out how the way to hack it (but it will be ugly)
> * removing the bindings and see what the user demand is

Given above: either of these seems reasonable. I reckon we timebox the first approach and if we can't make it work: go for the second.

**Comment by fxcoudert** [comment]
>  I reckon we timebox the first approach and if we can't make it work: go for the second.

I haven't been able to make a hack work (for a two-formula build: subversion itself in one formula, java bindings in another). If someone can step in, please do.

**Comment by fxcoudert** [comment]
I left 6 weeks for others to propose a different solution. I made another try at it myself, and failed. The formula is such a hack, it is ridiculous… I'd be actually tempted to remove the other bindings as well, given they require so many patches and hacks to just build.

So I am proposing again, per @MikeMcQuaid comment above, that we remove the java bindings.

**Review by SMillerDev — DISMISSED** [review]

**Review by iMichka — DISMISSED** [review]

**Comment by fxcoudert** [comment]
Previous run was fine, and produced bottles on all runners, but somehow the testbot cannot pick them up. Re-running.

**Pull request closed without merging**